src/zenml/zen_server/deploy/deployer.py

```diff
@@ -418,3 +418,34 @@ def get_server_logs(
         return provider.get_server_logs(
             server.config, follow=follow, tail=tail
         )
+
+    def get_deployment_outputs(
+        self,
+        server_name: str,
+        output_name: Optional[str] = None,
+    ) -> Dict[str, str]:
+        """Get the outputs of a ZenML server deployment.
+
+        Args:
+            server_name: The server deployment name.
+
+        Returns:
+            The outputs of the server deployment.
+        """
+        # this will also raise ServerDeploymentNotFoundError if the server
+        # does not exist
+        server = self.get_server(server_name)
+
+        provider_name = server.config.provider.value
+        provider = self.get_provider(server.config.provider)
+
+        # if the provider does not implement this method, we return an empty
+        # dict
+        if not hasattr(provider, "get_deployment_outputs"):
+            return {}
```

Comment:
Would it be worth to change this try/except and do some sort of logging that provider doesn't have `get_deployment_outputs`
```suggestion
        if not hasattr(provider, "get_deployment_outputs"):
            return {}
```